text/0240-app-telemetry-schema-vision-roadmap.md

```diff
@@ -0,0 +1,293 @@
+existing Telemetry Schema v1.1. It offers a way to import semantic convention
+registries and to define resource attributes, the properties of the
+instrumentation library, and the telemetry signals an application or library can
+produce. The existing versioning mechanism defined in version 1.1 remains
+unchanged. A Telemetry Schema can inherit from one or several other telemetry
+schemas (as explained in the previous section). The root schema is usually the
+official Telemetry Schema, referencing the official OpenTelemetry Semantic
+Convention Registry. The last schema (leaf) in this hierarchy is called the
+Application Telemetry Schema, as it defines all the signals a particular
+application produces. Although there is no direct lineage between these systems,
+a similar approach was designed and deployed by Facebook to address the same
+type of problem but in a proprietary context (refer to this
+[positional paper](https://research.facebook.com/publications/positional-paper-schema-first-application-telemetry/)
+for more information).
+
+The following diagram illustrates the relationships between these concepts and
+the main changes from Telemetry Schema v1.1 to v1.2.
+
+![Telemetry Schema](./img/0240-telemetry-schema-v1.1-to-v1.2.png)
+
+> Note 1: Each signal definition, where possible, reuses the existing syntax and
+> semantics defined by the semantic conventions. Each signal definition is also
+> identified by a unique name (or ID), making schemas easy to traverse, validate
+> , and diff.
+>
+> Note 2: This hierarchy of telemetry schemas helps large organizations in
+> collaborating on the Application Telemetry Schema. It enables different
+> aspects of an Application Telemetry Schema to be managed by various teams.
+>
+> Note 3: For all the elements that make up the Application Telemetry Schema, a
+> general mechanism of annotation or tagging will be integrated in order to
+> attach additional traits, characteristics, or constraints, allowing vendors
+> and companies to extend the definition of concepts defined by OpenTelemetry.
+>
+> Note 4: Annotations and Tags can also be employed to modify schemas for
+> diverse audiences. For example, the public version of a schema can exclude all
+> signals or other metadata labeled as private. Similarly, elements can be
+> designated as exclusively available for beta testers. These annotations can
+> also identify attributes as PII (Personally Identifiable Information), and
+> privacy policy enforcement can be implemented at various levels (e.g., in the
+> generated client SDK or in a proxy).
+
+Several OTEPs will be dedicated to the precise definition of the structure and
+the format of Telemetry Schema v1.2. The rules for resolving overrides
+(inheritance), external references, and conflicts will also be described in
+these OTEPs. See the Roadmap section for a comprehensive list of these OTEPs.
+
+## Resolved Telemetry Schema
```

Comment:
It is unclear to me what the scope of the resolved schema is. Is it a combination of all possible schemas used in an enterprise? Or a strict subset used in a specific application? If the latter, how does that work with oss instrumentation libraries that could refer to arbitrary schemas and are not uniformly discoverable?

Reply:
The difference is we're not trying to obtain an exhaustive schema for a given application. An application produces one or more datasets, eg an RPC instrumentation can produce RPC metrics dataset. We want each dataset to have a well defined schema, which is an easier problem. Then the question of all datasets for an app is a discovery problem, which can be solved via search, especially if the app/workload identity is semantically annotated in the dataset schemas. Eg find all metrics/logs/etc with attribute service=myservice.

Reply:
Added a new section discussing in more details these problematics.